crypto/cms/cms_env.c

```diff
@@ -1288,6 +1288,8 @@ int cms_pkey_get_ri_type(EVP_PKEY *pk)
     /* Check types that we know about */
     if (EVP_PKEY_is_a(pk, "DH"))
         return CMS_RECIPINFO_AGREE;
+    else if (EVP_PKEY_is_a(pk, "DHX"))
+        return CMS_RECIPINFO_AGREE;
```

Comment:
I am not convinced the `X9.42 DH` string is equivalent to one or more of the `DH` and `DHX` strings. And it's missing.

Reply:
X9.42 DH is the same as DHX. And it is handled in EVP_PKEY_is_a() explicitly for non-legacy keys.